commec/regulation/containers.py

```diff
@@ -0,0 +1,383 @@
+                         "It has been assigned to TaxID 0.", error_id)
+            non_empty = [("0", AccessionFormat.TAXID)]
+
+        object.__setattr__(self, "code", str(non_empty[0][0]))
+        object.__setattr__(self, "type", non_empty[0][1])
+
+    def get(self) -> tuple[str, str]:
+        """Return (code, type)."""
+        return self.code, self.type
+
+    def __str__(self):
+        return f"{self.code}"
+
+    def __repr__(self):
+        return f"{self.type}:{self.code}"
+
+@dataclass
+class Region:
+    """
+    Name and Acronym for regional information as either 
+    a country, or wider a organisation:
+    e.g. European Union, EU
+    e.g. New Zealand, NZ
+    """
+    name : str = ""
+    acronym : str = ""
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.name + " ["+self.acronym+"]"
+
+class ListMode(StrEnum):
+    """
+    Describes how a list is interpreted, based on regional context.
+    When a Taxonomy ID is identified as part of a list, whether or not we treat
+    that taxid as regulated or not will depend on the following mode.
+    
+    * COMPLIANCE - All taxids from this list are regulated.
+    * CONDITIONAL_NUM - Only mark taxid as regulated if it appears in more than 1 list.
+    * COMPLIANCE_WARN - Mark Taxid as only a WARNING.
+    * IGNORE - Ignore this list entirely.
+    """
+    COMPLIANCE = "Compliance"
+    CONDITIONAL_NUM = "Conditional Compliance"
+    COMPLIANCE_WARN = "Comply with Warning"
+    IGNORE = "Ignored"
+
+@dataclass
+class RegulationList:
+    """
+    Contains the name, acronym, url, and affected regions for a regulated list.
+    """
+    name : str = ""
+    acronym : str = ""
+    url : str = ""
+    regions : list[Region] = field(default_factory=list[Region])
+    status : ListMode = field(default_factory=ListMode)
+
+    def __str__(self):
+        regions_text = ""
+        for r in self.regions:
+            regions_text += str(r) + ", "
+        regions_text = regions_text[:-2]
+        return f"[{self.acronym}] {self.name} - {regions_text}\n({self.url})"
+    
+    def __eq__(self, value):
+        if (self.name != value.name or
+            self.url != value.url or 
+            set(self.regions) != set(value.regions) or
+            self.acronym != value.acronym):
+            return False
+        else:
+            return True
+
+@dataclass
+class TaxidRegulation:
+    """
+    Container for regulatory information of a given taxid from
+    a specific list. Structure data that is an expected output from
+    the regulations module.
+    """
+    category : str = ""
+    name : str = ""
+    notes : str = ""
+    derived_from : str = ""
+    preferred_taxonomy_name : str = ""
+    other_taxonomy_name : str = ""
+    tax_id : int = 0
+    list_acronym : str = ""
+    target : str = ""
+    hazard_group : str = ""
+    uniprot : str = ""
+    genbank_protein : str = ""
```

Comment:
I think `commec` can ignore some of these columns (e.g. notes, hazard_group, uniprot, genbank) for the moment, and would suggest we remove from the class